spec.emu

```diff
@@ -145,6 +145,84 @@ contributors: Google, Ecma International
 <emu-clause id="sup-temporal-calendar-abstract-ops">
   <h1>Abstract Operations</h1>
 
+  <emu-clause id="sup-temporal-calendardateadd" type="implementation-defined abstract operation">
+    <h1>
+      CalendarDateAdd (
+        _calendar_: a calendar type,
+        _isoDate_: an ISO Date Record,
+        _duration_: a Date Duration Record,
+        _overflow_: ~constrain~ or ~reject~,
+      ): either a normal completion containing an ISO Date Record or a throw completion
+    </h1>
+    <dl class="header">
+      <dt>description</dt>
+      <dd>
+        It adds _dateDuration_ to _isoDate_ using the years, months, and weeks reckoning of _calendar_.
+        If addition of years or months results in a nonexistent date, depending on _overflow_ it will be coerced to an existing date or the operation will throw.
+      </dd>
+    </dl>
+    <p>This definition supersedes the definition provided in TemporalCalendarDateAdd.</p>
+    <p>It performs the following steps when called:</p>
+    <emu-alg>
+      1. If _calendar_ is *"iso8601"*, then
+        1. Let _intermediate_ be BalanceISOYearMonth(_isoDate_.[[Year]] + _duration_.[[Years]], _isoDate_.[[Month]] + _duration_.[[Months]]).
+        1. Set _intermediate_ to ? RegulateISODate(_intermediate_.[[Year]], _intermediate_.[[Month]], _isoDate_.[[Day]], _overflow_).
+        1. Let _d_ be _intermediate_.[[Day]] + _duration_.[[Days]] + 7 × _duration_.[[Weeks]].
+        1. Let _result_ be BalanceISODate(_intermediate_.[[Year]], _intermediate_.[[Month]], _d_).
+      1. Else,
+        1. NOTE: The algorithm when _calendar_ is not *"iso8601"* is implementation-defined, but all calendars follow the general steps given here, which is a generalization of the precise algorithm specified above for *"iso8601"*.
+        1. Let _calendarDate_ be CalendarISOToDate(_calendar_, _isoDate_).
+        1. Add _duration_.[[Years]] to _calendarDate_.
+        1. (This step only matters for lunisolar calendars.) If _calendarDate_.[[MonthCode]] is a leap month that doesn't exist in the year, then:
```

Comment:
This is the step we were discussing in the Matrix channel. Basically, we think it makes sense to do a year-month constrain/reject step after adding the years and before adding the months. Otherwise, you are not certain which month to start counting from when you add the months.